### PR TITLE
Feature/apps 2543 edit org title description

### PR DESCRIPTION
**Pull request by solomontang**
# Adds
- Edit organization title and description (https://www.figma.com/file/PupCqZk6vLO6SdvyHnv4cv/Organizations?node-id=510%3A349)

image
image

### CODE REVIEW

File: src/containers/org/components/EditDescriptionBody.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            <Input
+                id="org-description"
+                onChange={(e :SyntheticEvent<HTMLInputElement>) => setDescription(e.currentTarget.value)}
+                value={description} />
+          </div>
+        </StyledStack>
+        <Button
+            color="primary"
+            fullWidth
+            isLoading={requestState === RequestStates.PENDING}
```
Comment (anzioka) [inline src/containers/org/components/EditDescriptionBody.js:86]: no change required... `isPending` is now in `ReduxUtils`

File: src/containers/org/components/OrgActionButton.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  const handleCloseMenu = () => {
+    dispatch({ type: 'closeMenu' });
+  };
+
+  const handleOpenDescription = () => {
+    dispatch({ type: 'openDescription' });
+  };
+
+  const handleCloseDescription = () => {
+    dispatch({ type: 'closeDescription' });
```
Comment (anzioka) [inline src/containers/org/components/OrgActionButton.js:68]: constantify these types?

File: src/containers/orgs/OrgsReducer.js [hunk 3]
```diff
@@ -258,6 +261,27 @@ export default function reducer(state :Map = INITIAL_STATE, action :Object) {
...
+          const { organizationId, title, description } = action.value;
+          const currentOrg :Organization = state.getIn([ORGS, organizationId]);
+          const updatedOrg = currentOrg.toImmutable().merge({
+            description,
+            title,
+          });
+
+          return state
+            .setIn([EDIT_ORGANIZATION_DETAILS, REQUEST_STATE], RequestStates.SUCCESS)
+            .setIn([ORGS, organizationId], new OrganizationBuilder(updatedOrg).build());
```
Comment (anzioka) [inline src/containers/orgs/OrgsReducer.js:279]: shouldn't this be converted back to immutable?

Comment (solomontang) [reply src/containers/orgs/OrgsReducer.js:279]: an organization is a class object and is not an immutable collection

**Review by anzioka — APPROVED** [review]

**Review by UnsungHero97 — APPROVED** [review]

File: src/containers/org/sagas/index.js [hunk 4]
```diff
@@ -4,7 +4,8 @@
 
 export * from './addRoleToOrganization';
 export * from './createNewOrganization';
+export * from './editMetadata';
```
[the remark this answers is not in the document]
Comment (solomontang) [reply src/containers/org/sagas/index.js:7]: the imports are just being reordered. It's not new in this PR

File: src/containers/orgs/OrgsReducer.js [hunk 5]
```diff
@@ -258,6 +261,27 @@ export default function reducer(state :Map = INITIAL_STATE, action :Object) {
...
+        REQUEST: () => state
+          .setIn([EDIT_ORGANIZATION_DETAILS, REQUEST_STATE], RequestStates.PENDING)
+          .setIn([EDIT_ORGANIZATION_DETAILS, action.id], action),
+        SUCCESS: () => {
+          const { organizationId, title, description } = action.value;
+          const currentOrg :Organization = state.getIn([ORGS, organizationId]);
+          const updatedOrg = currentOrg.toImmutable().merge({
+            description,
+            title,
+          });
```
[the remark this answers is not in the document]
Comment (solomontang) [reply src/containers/orgs/OrgsReducer.js:275]: I thought about doing this but the updateOrganizationDescription/Title APIs do not return any data in their responses. I can make a PR to update lattice-sagas to return the requested changes. 

In the end, it doesn't change what the user behavior would be if they encounter either error. The modal does not close upon failure so the user is free to try again immediately without losing either requested change. That being said, we should definitely get designs for how to notify users of submission failures.

**Pull request merged**